### PR TITLE
UNIFY-616-specPattern

### PR DESCRIPTION
**Pull request by jaffrepaul**
https://cypress-io.atlassian.net/browse/UNIFY-616

### CODE REVIEW

File: content/api/plugins/before-run-api.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ module.exports = (on, config) => {
     //     // ... more specs
     //   ],
     //   specPattern: [
-    //     '**/*_spec.js'
+    //     '**/*.cy.{js,jsx,ts,tsx}'
```
Comment (cowboy) [inline content/api/plugins/before-run-api.md:77]: The `specPattern` won't default to that value if the `specs` array contains a spec like `relative: 'cypress/integration/login_spec.js'`. This example might need to be double-checked.

File: content/guides/component-testing/introduction.md [hunk 2]
```diff
@@ -114,10 +114,11 @@ look like this:
 
 ```js
 {
-  testFiles: "cypress/integration/*.spec.js",
+  e2e: {
```
Comment (cowboy) [inline content/guides/component-testing/introduction.md:117]: Shouldn't this file actually be deleted? See https://github.com/cypress-io/cypress-documentation/blob/unification/netlify.toml#L56-L60

File: content/guides/getting-started/component-framework-configuration.md [hunk 3]
```diff
@@ -51,17 +51,15 @@ const { devServer } = require('@cypress/react/plugins/react-scripts')
 {
   component: {
     devServer,
-    componentFolder: 'src',
-    testFiles: '**/*.test.{js,ts,jsx,tsx}'
+    specPattern: '**/*.cy.{js,jsx,ts,tsx}'
```
Comment (cowboy) [inline content/guides/getting-started/component-framework-configuration.md:54]: It doesn't entirely make sense to recommend that the user specify a `specPattern` option with the default value. Also, the text before the example doesn't match the pattern. 

Perhaps change the text in the paragraph above to say something like this (and maybe link `specPattern` to the place in the configuration guide where it is explained):

> By default, Cypress looks for spec files anywhere in your project with an extension of either `.cy.js`, `.cy.jsx`, `.cy.ts`, or `.cy.tsx`. However, you can change this behavior with a custom `specPattern` value. In the following example, we've configured Cypress to look for spec files with those same extensions, but only in the `src` folder or any of its subdirectories.

And then change the `specPattern` to `'src/**/*.cy.{js,jsx,ts,tsx}'`

This comment applies to all 5 examples on this page.

File: content/guides/references/configuration.md [hunk 4]
```diff
@@ -238,7 +235,7 @@ object:
...
 | `setupNodeEvents` | `null`                     | Function in which node events can be registered and config can be modified. Takes the place of the (deprecated) plugins file. [Please read the notes for examples on using this.](#setupNodeEvents) |
 | `supportFile`     | `cypress/support/index.js` | Path to file to load before test files load. This file is compiled and bundled. (Pass `false` to disable)                                                                                           |
-| `testFiles`       | `**/*.*`                   | A String or Array of glob patterns of the test files to load                                                                                                                                        |
+| `specPattern`     | `**/*.cy.{js,jsx,ts,tsx}`  | A String or Array of glob patterns of the test files to load                                                                                                                                        |
```
Comment (cowboy) [inline content/guides/references/configuration.md:238]: Please add a note here to this effect: https://www.notion.so/cypressdx/testFiles-specPattern-glob-patterns-295d545a0af942ebb5b71eda2a6d422b#1a893de703d14709bc159c17559d2a97

File: content/guides/references/migration-guide.md [hunk 5]
```diff
@@ -366,8 +366,7 @@ required for component testing
 ```json
 {
   "experimentalComponentTesting": true,
-  "componentFolder": "src",
-  "testFiles": "**/*spec.{js,jsx,ts,tsx}"
+  "specPattern": "**/*spec.{js,jsx,ts,tsx}"
```
Comment (cowboy) [inline content/guides/references/migration-guide.md:369]: Revert all changes to the Migration Guide, the `specPattern` change won't be retroactively applied to older versions of Cypress

**Review by cowboy — CHANGES_REQUESTED** [review]

File: content/guides/references/migration-guide.md (hunk 5 above)
Comment (jaffrepaul) [reply content/guides/references/migration-guide.md:369]: Yep 👍 I had been filtering out this page as to not change it. Must have slipped in there during one of the project wide searches.

File: content/guides/component-testing/introduction.md (hunk 2 above)
Comment (jaffrepaul) [reply content/guides/component-testing/introduction.md:117]: Yeah you're right. It's still in `unification`. IDK if a previous PR was to handle that. I'll pull it from here.

File: content/api/plugins/before-run-api.md (hunk 1 above)
Comment (jaffrepaul) [reply content/api/plugins/before-run-api.md:77]: Agreed. I wasn't certain on this one and forgot to add that note to the PR. I'll check it out when we get the dev build.

File: content/guides/getting-started/component-framework-configuration.md [hunk 6]
```diff
@@ -36,10 +36,11 @@ Configure the dev server to use the same Webpack configuration used by Create
 React App. You can do this using the `react-scripts` plugin provided by the
 `@cypress/react` module.
 
-Also, you need to tell Cypress where to find your component tests. The following
-example configuration assumes that all the test files are somewhere in the `src`
-folder and end with a `.test.js`, `.test.jsx`, `.test.ts` or `.test.tsx`
-extension.
+By default, Cypress looks for spec files anywhere in your project with an
```
Comment (cowboy) [inline content/guides/getting-started/component-framework-configuration.md:39]: You need to make an equivalent change to the text preceding the 4 other examples where `specPattern` is used. People are most likely going to jump to the example that is specific to their needs, and won't see this one. Double-check the examples at https://github.com/cypress-io/cypress-component-testing-examples to figure out what the best non-default `specPattern` value should be for each example.

File: content/guides/references/configuration.md [hunk 7]
```diff
@@ -232,13 +229,13 @@ object:
...
+| `devServerConfig` | `null`                     | Optional dev server configuration. Config options will be determined by the dev server. [Please read the notes for examples on using this.](#devServer-devServerConfig)                                                                      |
+| `setupNodeEvents` | `null`                     | Function in which node events can be registered and config can be modified. Takes the place of the (deprecated) plugins file. [Please read the notes for examples on using this.](#setupNodeEvents)                                          |
+| `supportFile`     | `cypress/support/index.js` | Path to file to load before test files load. This file is compiled and bundled. (Pass `false` to disable)                                                                                                                                    |
+| `specPattern`     | `**/*.cy.{js,jsx,ts,tsx}`  | A String or Array of glob patterns of the test files to load. <br><br>\*Note on matching: component tests will be found using the component.specPattern value but any tests found matching the e2e.specPattern value will be **excluded.\*** |
```
Comment (cowboy) [inline content/guides/references/configuration.md:238]: This doesn't seem to be rendering correctly. Can you please double-check the formatting?

**Review by cowboy — CHANGES_REQUESTED** [review]

File: content/guides/getting-started/component-framework-configuration.md (hunk 6 above)
Comment (jaffrepaul) [reply content/guides/getting-started/component-framework-configuration.md:39]: @cowboy We can chat when you review the latest if need be. There's some nuance where I kept the Next.js (Webpack 4) example mostly in tact while it's testing an index component in the `pages` directory. But we also have a note in the Nuxt section about that sort of content being better used as part of e2e tests. I left these in and we can talk through if changes are required to make things more clear.

File: content/guides/getting-started/component-framework-configuration.md [hunk 8]
```diff
@@ -401,17 +401,15 @@ const { getWebpackConfig } = require('nuxt')
         webpackConfig,
       })
     },
-    componentFolder: 'components',
-    testFiles: '**/*.spec.js'
+    specPattern: 'components/**/*.cy.{js,jsx,ts,tsx}'
```
Comment (cowboy) [inline content/guides/getting-started/component-framework-configuration.md:404]: Like with Next.js, I think for Nuxt, we'll want to use a `specPattern` like `'{pages,components}/**/*.cy.{js,jsx,ts,tsx}'`.

File: content/guides/references/configuration.md [hunk 9]
```diff
@@ -232,13 +229,13 @@ object:
...
+| `devServerConfig` | `null`                     | Optional dev server configuration. Config options will be determined by the dev server. [Please read the notes for examples on using this.](#devServer-devServerConfig)                                                                        |
+| `setupNodeEvents` | `null`                     | Function in which node events can be registered and config can be modified. Takes the place of the (deprecated) plugins file. [Please read the notes for examples on using this.](#setupNodeEvents)                                            |
+| `supportFile`     | `cypress/support/index.js` | Path to file to load before test files load. This file is compiled and bundled. (Pass `false` to disable)                                                                                                                                      |
+| `specPattern`     | `**/*.cy.{js,jsx,ts,tsx}`  | A String or Array of glob patterns of the test files to load. <br><br>\*Note on matching: component tests will be found using the `component.specPattern` value but any tests found matching the `e2e.specPattern` value will be **excluded.** |
```
Comment (cowboy) [inline content/guides/references/configuration.md:238]:
```suggestion
| `specPattern`     | `**/*.cy.{js,jsx,ts,tsx}`  | A String or Array of glob patterns of the test files to load. <br><br>Note that any files found matching the `e2e.specPattern` value will be automatically **excluded.** |
```

**Review by cowboy — CHANGES_REQUESTED** [review]

**Review by cowboy — APPROVED** [review]

**Pull request merged**